automation/src/test/scala/org/broadinstitute/dsde/firecloud/api/FireCloudClient.scala

```diff
@@ -0,0 +1,113 @@
+  implicit val ec: ExecutionContextExecutor = ExecutionContext.global
+
+  val mapper = new ObjectMapper()
+  mapper.registerModule(DefaultScalaModule)
+
+  // TODO: Is this okay here or should it be in a separate object and explicitly imported?
+  implicit class JsonStringUtil(s: String) {
+    def fromJsonMapAs[A](key: String): Option[A] = parseJsonAsMap.get(key)
+
+    def parseJsonAsMap[A]: Map[String, A] = mapper.readValue(s, classOf[Map[String, A]])
+  }
+
+  private def makeAuthHeader(token: AuthToken): Authorization = {
+    headers.Authorization(OAuth2BearerToken(token.value))
+  }
+
+  private def sendRequest(httpRequest: HttpRequest): HttpResponse = {
+    Await.result(Http().singleRequest(httpRequest), 5.minutes)
+  }
+
+  private def handleResponse(response: HttpResponse): String = {
+    response.status.isSuccess() match {
+      case true =>
+        val byteStringSink: Sink[ByteString, Future[ByteString]] = Sink.fold(ByteString("")) { (z, i) => z.concat(i) }
+        val entityFuture = response.entity.dataBytes.runWith(byteStringSink)
+        Await.result(entityFuture, 100.millis).decodeString("UTF-8")
+      case _ =>
+        val byteStringSink: Sink[ByteString, Future[ByteString]] = Sink.fold(ByteString("")) { (z, i) => z.concat(i) }
+        val entityFuture = response.entity.dataBytes.runWith(byteStringSink)
+        throw new APIException(Await.result(entityFuture, 100.millis).decodeString("UTF-8"))
+    }
+  }
+
+  def parseResponse(response: HttpResponse): String = {
+    response.status.isSuccess() match {
+      case true =>
+        val byteStringSink: Sink[ByteString, Future[ByteString]] = Sink.fold(ByteString("")) { (z, i) => z.concat(i) }
+        val entityFuture = response.entity.dataBytes.runWith(byteStringSink)
+        Await.result(entityFuture, 100.millis).decodeString("UTF-8")
+      case _ =>
+        val byteStringSink: Sink[ByteString, Future[ByteString]] = Sink.fold(ByteString("")) { (z, i) => z.concat(i) }
+        val entityFuture = response.entity.dataBytes.runWith(byteStringSink)
+        throw new APIException(Await.result(entityFuture, 100.millis).decodeString("UTF-8"))
+    }
+  }
+
+  private def requestWithJsonContent(method: HttpMethod, uri: String, content: Any, httpHeaders: List[HttpHeader] = List())(implicit token: AuthToken): String = {
+    val req = HttpRequest(method, uri, List(makeAuthHeader(token)), HttpEntity(ContentTypes.`application/json`, mapper.writeValueAsString(content)))
+    handleResponse(sendRequest(req))
+  }
+
+  def postRequestWithMultipart(uri:String, name: String, content: String)(implicit token: AuthToken): String = {
+    // val cont = ContentType.parse("text/tab-separated-values")
+    //val ranges = coalesceRanges(iRanges).sortBy(_.start)
+    //val formD = Multipart.ByteRanges(HttpEntity(ByteString("...")))
+    val formData = Multipart.FormData {
+      Source {
+        List (
+          Multipart.FormData.BodyPart.apply(name, HttpEntity(ByteString(content)))
```

Comment:
Instead of `Source(List(Multipart.FormData.BodyPart(...)))` just `Source.single(Multipart.FormData.BodyPart(...)))`. 

Can also omit the `.apply`